src/nodes.coffee

```diff
@@ -687,12 +686,20 @@ exports.Call = class Call extends Base
 
     fragments = []
     if this instanceof SuperCall
-      preface = @superReference(o) + ".call(#{@superThis(o)}"
-      if compiledArgs.length then preface += ", "
+      preface = @superReference o
+      if preface isnt 'super'
+        preface += ".call(#{@superThis o}"
+        preface += ', ' if compiledArgs.length
+      else
+        preface += '('
       fragments.push @makeCode preface
     else
-      if @isNew then fragments.push @makeCode 'new '
+      complex = @variable.isComplex()
+      if @isNew
+        fragments.push @makeCode 'new '
+        fragments.push @makeCode '(' if complex
```

Comment:
This wraps complex `new` targets in parenthesis (e.g. `new (a())`) to ensure there are no errors from calling constructors without `new` (e.g. [this test](https://github.com/jashkenas/coffeescript/pull/4330/files#diff-1fc895c05f3855ef64b85685f62803acR469) was throwing that error without this change).